File: 2470-removing-stars-from-a-string/removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {
        StringBuilder sb = new StringBuilder();
        char[] str = s.toCharArray();
 
        for (int i = 0; i < s.length(); i++)
        {
           if (str[i] == '*')
           {
              if (!sb.isEmpty())
              {
                  int lastPos = sb.length() - 1;
                  sb.deleteCharAt(lastPos);
              }
           }
           else
             sb.append(str[i]);
        }

        return sb.toString();
    }
}

File: 2470-removing-stars-from-a-string/removing-stars-from-a-string.cpp
class Solution {
public:
    string removeStars(string s) {
        
        string res;
 
        int n = s.size();
        for (int i = 0; i < n; i++)
        {
           if (s[i] == '*')
           {
              if (!res.empty())
                res.pop_back();
           }
           else
             res.push_back(s[i]);
        }

        return res;
    }
};